### PR TITLE
Port over oss-test-infra prow config and jobs.

### PR DESCRIPTION
**Pull request by kvmware**
Fixes https://github.com/knative/test-infra/issues/2982

Please read through the commits. I tried to make them very easy and bite size to review.
Note: I moved all prow related things from `config/prow` to `./prow`. The testgrid and gubernator (that needs to be removed) I left in `config/prow` for now. This might be confusing for someone who comes from kubernetes/test-infra as usually the root `prow` directory is used for prow code not yaml, but for people joining our team and a little documentation this new placement will make much more sense.

EDIT: should fix https://github.com/knative/test-infra/issues/3060 as well.

### CODE REVIEW

File: prow/create-build-cluster.sh [hunk 1]
```diff
@@ -1,23 +1,46 @@
 #!/usr/bin/env bash
+
+# Copyright 2020 The Knative Authors
```
Comment (chizhg) [inline prow/create-build-cluster.sh:3]: Year should be 2022. Ditto for elsewhere that are newly licensed.

File: prow/config.sh [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 
-# Copyright 2021 The Knative Authors
+# Copyright 2022 The Knative Authors
```
Comment (chizhg) [inline prow/config.sh:3]: I think the rule is if we are only modifying the file, no need to change the copyright year.

**Review by chizhg — COMMENTED** [review]
Could you fix the lint test?

File: prow/config.sh (hunk 2 above)
Comment (kvmware) [reply prow/config.sh:3]: Ok so only when we are changing from `Google` to `Knative Authors`?

Comment (chizhg) [reply prow/config.sh:3]: Yep

**Pull request merged**